### ISSUE TITLE
Unexpected incompatibility when using `TypeVarTuple`

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pyright#3591]
## Environment data

-   Language Server version: v2022.6.21 Pre-Release
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.12 Anaconda

## Code Snippet

```python
from __future__ import annotations

from typing import Protocol
from typing_extensions import TypeVarTuple, Unpack

Values = TypeVarTuple("Values")

def f(*args: Unpack[Values]) -> tuple[Unpack[Values]]:
    ...

class FP(Protocol):
    def __call__(self, v0: int, v1: int, /) -> tuple[int, int]:
        ...

fp: FP = f
```

## Expected behavior

No errors/warnings.

## Actual behavior

![image](https://user-images.githubusercontent.com/15389744/174531071-fd97f002-3951-42ec-932e-93ab7a1e0b50.png)

**Comment by erictraut** [comment]
This pattern isn't supported for TypeVarTuple currently. I'm not sure if it was intended to be supported by the authors of PEP 646. None of their samples seem to anticipate this usage.

I don't see any good reason not to support it from a type safety perspective. However, it does seem extremely niche, and it would require adding a bunch of new special-case logic in a piece of code that is already one of the most complex in all of pyright. 

I'll think about this more, but I may decide to reject this feature request for now and wait to see if other pyright users are also interested in this niche use case.

**Comment by erictraut** [comment]
I'm going to reject this feature request for now. We may revisit this in the future if there is sufficient demand. As I mentioned above, it's not clear that the authors of PEP 646 anticipated this usage.
